pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -211,10 +211,6 @@ func MergeRuntimeContainers(runtimeContainer *v1.Container, predictorContainer *
 	if mergedContainer.Name == "" {
 		mergedContainer.Name = runtimeContainerName
 	}
-
-	// Strategic merge patch will replace args but more useful behaviour here is to concatenate
-	mergedContainer.Args = append(append([]string{}, runtimeContainer.Args...), predictorContainer.Args...)
```

Comment:
Looks like tests are failing because test cases expect that runtime container args need to exist in the merged container args. 

Should we consider adding a flag to turn this on/off?

Reply:
Ok CI passing now. So basically if there's an overlap, we don't merge them since the merge result will be incorrect and contain duplicates.